index.js

```diff
@@ -105,7 +105,11 @@ function read(file) {
 
 function extend(a, b) {
   Object.keys(b).forEach(function(key) {
-    if (!a.hasOwnProperty(key)) a[key] = b[key];
+    if (!a.hasOwnProperty(key)) {
+      a[key] = b[key];
+    } else if (b[key].constructor && b[key].constructor === Object) {
```

Comment:
This is a little stricter than we _probably_ want to be. I can't think of an awesome use-case, but maybe you want to mix in an instance object (i.e. from `new Foo()`) instead? In virtue of the fact that you're checking the constructor property, that won't work. Would recommend using the [component/type library](https://github.com/component/type) or something like...

```js
var isObj = b && ({}).toString.call(b) === '[object Object]';
```